quasar-core/src/main/java/co/paralleluniverse/fibers/Fiber.java

```diff
@@ -2115,6 +2117,15 @@ public static ByteArraySerializer getFiberSerializer(boolean includeThreadLocals
 
     private static class FiberSerializer extends Serializer<Fiber<?>> {
         private final boolean includeThreadLocals;
+        private final Map<Class<? extends Fiber>, FieldSerializer<Fiber<?>>> fieldSerializers = new HashMap<>();
+
+        private FieldSerializer<Fiber<?>> getFieldSerializer(Kryo kryo, Class<? extends Fiber> fiberClass) {
+            FieldSerializer<Fiber<?>> serializer = fieldSerializers.get(fiberClass);
+            if(serializer != null) return serializer;
+            serializer = new FieldSerializer<>(kryo, fiberClass);
+            fieldSerializers.putIfAbsent(fiberClass, serializer);
```

Comment:
You _must_ be assuming that this `Map` will be accessed by multiple threads simultaneously, because otherwise you could just use:
```java
fieldSerializers.put(fiberClass, serializer);
```
here, seeing as `get()` has already returned `null`. [But `HashMap.putIfAbsent()` is not atomic.](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#putIfAbsent-K-V-):
> The default implementation makes no guarantees about synchronization or atomicity properties of this method. Any implementation providing atomicity guarantees must override this method and document its concurrency properties.